lib/workingsetcache/cache.go

```diff
@@ -355,13 +373,23 @@ func (c *Cache) Reset() {
 	curr.UpdateStats(&cs)
 	updateCacheStatsHistory(&c.csHistory, &cs)
 	curr.Reset()
-	// Reset the mode to `split` in the hope the working set size becomes smaller after the reset.
+	// Reset the mode to `split` in order to properly reset background workers.
+	mode := c.mode.Load()
 	c.mode.Store(split)
 	c.copiedFromPrev.Store(0)
+	if mode != split {
```

Comment:
should we do this unconditionally to simplify the code? As I understand, caches are already reseted and its maxSize can be restored to nominal values anyway.

Reply:
It's better to conditionally allocate cache in order to reduce memory leak possibilities. Since curr could be referenced by Set or Get, it may add item into it after Reset call. And it will leak memory ( not track-able by golang memory profiler). I'm going to address it at the separate PR.